### ISSUE TITLE
Full width headings inside group block ignores padding

### ISSUE THREAD

**Issue by carolinan** [WordPress/twentytwentyone#729]
**Describe the bug**
When a group block has a background color, it has some additional padding:
```
.editor-styles-wrapper .wp-block-group.has-background {
    padding: var(--global--spacing-vertical);
}
```
When a heading is added inside the group block, and is set to be _full width_,
the heading ignores this padding and moves right up to the edge of the group block.

**To Reproduce**
1. Add a group block with a background color.
2. Add a heading block.
3. Set the heading to full width.

**Expected behavior**
Perhaps this is working as intended? I don't think it looks as good when text ignores the padding.

**Screenshots**

![full-width-heading](https://user-images.githubusercontent.com/7422055/97449752-4e70b300-1932-11eb-8dc7-0e8924bed85e.gif)

WordPress version: 5.6-beta2-49352
Does the website has Gutenberg plugin installed, or is it using the block editor that comes by default? "default"

**Comment by carolinan** [comment]
I forgot to test the front: It works well on the front, there is space between the heading and the edge of the block.

**Comment by carolinan** [comment]
This seems intentional, because I see this style in the editor:
```
.wp-block-group.has-background>.wp-block-group__inner-container>[data-align=full] {
    margin-left: -30px;
    width: calc(100% + 60px);
}
```
How do we want to handle it for this theme?

**Comment by kjellr** [comment]
This is working as expected for me. If it's set to be full width, it would be full width. Same thing happens with a full-width image block inside of there for example. 

There's a case to be made for adding additional left/right padding to the full-width heading block itself so it doesn't 100% bump up against the edge, but that seems like something we'd want to go globally, just just in this case.

**Comment by carolinan** [comment]
Then it is not working as intended on the front, where the full width heading does not bump against the edge.

**Comment by kjellr** [comment]
> Then it is not working as intended on the front, where the full width heading does not bump against the edge.

Turned out this is an issue for _all_ full-width blocks when the group has a background, or when it has the "Frame" block style. Should be fixed in #742.
